packages/editor/src/pages/Editor/components/Main/Editor/ReactMonaco.tsx

```diff
@@ -41,8 +41,10 @@ export class ReactMonaco extends Component<IProps, IState> {
         this.clearAllModels();
       }
 
-      if (file.id !== prevProps.file.id) {
-        const newModel = this.getModel();
+      const isDifferentFile = file.id !== prevProps.file.id;
```

Comment:
*insert usual comment about how I feel*
```typescript
if (
  file.id !== prevProps.file.id || 
  file.language !== prevProps.file.language
) {

}
```
is cleaner than bringing them out to variables especially if they're only used once.